gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaUtils.java

```diff
@@ -89,11 +89,15 @@ public static List<KafkaPartition> getPartitions(State state) {
       if (!state.contains(KafkaUtils.getPartitionPropName(KafkaSource.PARTITION_ID, i))) {
         break;
       }
-      KafkaPartition partition = new KafkaPartition.Builder().withTopicName(topicName)
-          .withId(state.getPropAsInt(KafkaUtils.getPartitionPropName(KafkaSource.PARTITION_ID, i)))
-          .withLeaderId(state.getPropAsInt(KafkaUtils.getPartitionPropName(KafkaSource.LEADER_ID, i)))
-          .withLeaderHostAndPort(state.getProp(KafkaUtils.getPartitionPropName(KafkaSource.LEADER_HOSTANDPORT, i)))
-          .build();
+      KafkaPartition.Builder builder = new KafkaPartition.Builder().withTopicName(topicName)
+          .withId(state.getPropAsInt(KafkaUtils.getPartitionPropName(KafkaSource.PARTITION_ID, i)));
+      if (state.contains(KafkaUtils.getPartitionPropName(KafkaSource.LEADER_ID, i))) {
```

Comment:
I'd favor not copy pasting the `KafkaUtils.getPartitionPropName(KafkaSource.LEADER_ID, i)` but that is just me. Also, I assume that kafka utils cannot be null right?

Reply:
getPartitionPropName is a static method of KafkaUtils.